src/main/java/it/gov/pagopa/payment/service/qrcode/QRCodePreAuthServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package it.gov.pagopa.payment.service.qrcode;
+
+import it.gov.pagopa.payment.connector.rest.reward.RewardCalculatorConnector;
+import it.gov.pagopa.payment.connector.rest.reward.dto.AuthPaymentResponseDTO;
+import it.gov.pagopa.payment.connector.rest.reward.mapper.AuthPaymentRequestMapper;
+import it.gov.pagopa.payment.dto.mapper.TransactionInProgress2TransactionResponseMapper;
+import it.gov.pagopa.payment.dto.qrcode.TransactionResponse;
+import it.gov.pagopa.payment.enums.SyncTrxStatus;
+import it.gov.pagopa.payment.exception.ClientExceptionWithBody;
+import it.gov.pagopa.payment.model.TransactionInProgress;
+import it.gov.pagopa.payment.repository.TransactionInProgressRepository;
+import org.springframework.http.HttpStatus;
+
+public class QRCodePreAuthServiceImpl implements QRCodePreAuthService {
+
+  private final TransactionInProgressRepository transactionInProgressRepository;
+  private final TransactionInProgress2TransactionResponseMapper transactionInProgress2TransactionResponseMapper;
+  private final AuthPaymentRequestMapper authPaymentRequestMapper;
+  private final RewardCalculatorConnector rewardCalculatorConnector;
+
+  public QRCodePreAuthServiceImpl(TransactionInProgressRepository transactionInProgressRepository,
+      TransactionInProgress2TransactionResponseMapper transactionInProgress2TransactionResponseMapper,
+      AuthPaymentRequestMapper authPaymentRequestMapper, RewardCalculatorConnector rewardCalculatorConnector) {
+    this.transactionInProgressRepository = transactionInProgressRepository;
+    this.transactionInProgress2TransactionResponseMapper = transactionInProgress2TransactionResponseMapper;
+    this.authPaymentRequestMapper = authPaymentRequestMapper;
+    this.rewardCalculatorConnector = rewardCalculatorConnector;
+  }
+
+  @Override
+  public TransactionResponse relateUser(String trxCode, String userId) {
+    TransactionInProgress trx = transactionInProgressRepository.findByTrxCodeAndRelateUser(trxCode, userId);
```

Comment:
filter also by trxChargeDate as done in auth API

Reply:
let's not search filtering by userId, but explicitly test the equals returning 403 if not equals (in order to be aligned with auth behavior and DR)